src/avro/writer.rs

```diff
@@ -100,7 +100,30 @@ impl<W: Write> Writer<W> {
         } else {
             0
         };
+        self.append_inner(n, value)
+    }
+
+    /// Append, but useful for creating files that are less-robust to improve round-trip testing
+    ///
+    /// This is public but only for internal use, its API might change without warning.
+    #[doc(hidden)]
+    pub fn __append_for_testing<T: ToAvro>(
+        &mut self,
+        value: T,
+        include_codec: bool,
+    ) -> Result<usize, Error> {
+        let n = if !self.has_header {
+            let header = self.header_inner(include_codec)?;
+            let n = self.append_bytes(header.as_ref())?;
+            self.has_header = true;
+            n
+        } else {
+            0
+        };
+        self.append_inner(n, value)
```

Comment:
Can you explain a bit what is going on with `n` here? Does `append_inner` expect the size of the header, or what?

Reply:
I've changed the append_inner parameter to `bytes_so_far`, because that's what it means, happy to hear other colors for this variable